### PR TITLE
IBP-4149 add stock id from plating preparation in observation tab

### PR DESCRIPTION
**Pull request by cuenyad**
Please review.

Included:
https://github.com/IntegratedBreedingPlatform/Middleware/pull/921
https://github.com/IntegratedBreedingPlatform/Fieldbook/pull/945

Regards, Diego

### CODE REVIEW

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java [hunk 1]
```diff
@@ -79,6 +88,14 @@ File generate(
 
 		// Get all variables for the dataset
 		final List<MeasurementVariable> columns = this.getColumns(study.getId(), dataSet.getDatasetId());
+		if (dataSet.getDatasetTypeId().equals(DatasetTypeEnum.PLOT_DATA.getId())) {
+			final LotsSearchDto lotsSearchDto = new LotsSearchDto();
+			lotsSearchDto.setPlantingStudyIds(Arrays.asList(studyId));
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java:93]: @cuenyad  Is it filtering the by cancelled transactions?

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java [hunk 2]
```diff
@@ -182,6 +199,12 @@ File getReturnFile(final Study study, final List<File> files) throws IOException
 		return columns;
 	}
 
+	protected void addStockIdColumn(final List<MeasurementVariable> plotDataSetColumns) {
+		final Optional<MeasurementVariable>
+			designationColumn = plotDataSetColumns.stream().filter(measurementVariable -> measurementVariable.getName().equals("DESIGNATION")).findFirst();
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java:204]: same here, should not be better to compare by id using TermId ? @cuenyad

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java [hunk 3]
```diff
@@ -190,4 +213,16 @@ File getReturnFile(final Study study, final List<File> files) throws IOException
 	void setZipUtil(final ZipUtil zipUtil) {
 		this.zipUtil = zipUtil;
 	}
+
+	private MeasurementVariable buildStockIdColumn() {
```
Comment (clarysabel) [inline src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java:217]: You can send the TemId, VariableType and factor as parameters and make it reusable @cuenyad

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java]: Fixied

File: src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/ibp/api/java/impl/middleware/dataset/AbstractDatasetExportService.java]: Done.

**Pull request merged**